p2p/providers/peerstore_provider/rpc/provider.go

```diff
@@ -17,36 +17,44 @@ import (
 	"github.com/pokt-network/pocket/shared/modules/base_modules"
 )
 
-var _ peerstore_provider.PeerstoreProvider = &rpcPeerstoreProvider{}
+var (
+	_ modules.Submodule                    = &rpcPeerstoreProvider{}
+	_ peerstore_provider.PeerstoreProvider = &rpcPeerstoreProvider{}
+	_ rpcPeerstoreProviderFactory          = &rpcPeerstoreProvider{}
+)
+
+type rpcPeerstoreProviderOption func(*rpcPeerstoreProvider)
+type rpcPeerstoreProviderFactory = modules.FactoryWithOptions[peerstore_provider.PeerstoreProvider, rpcPeerstoreProviderOption]
 
-// TECHDEBT(#810): refactor to implement `Submodule` interface.
 type rpcPeerstoreProvider struct {
-	// TECHDEBT(#810): simplify once submodules are more convenient to retrieve.
-	base_modules.IntegratableModule
-	base_modules.InterruptableModule
+	base_modules.IntegrableModule
 
 	rpcURL    string
 	p2pCfg    *configs.P2PConfig
```

Comment:
I think this can actually be removed, it doesn't look like `rpcPeerstoreProvider#p2pCfg` is used anywhere.

```suggestion
```